streaming/src/main/scala/org/apache/gearpump/streaming/task/Task.scala

```diff
@@ -171,6 +171,8 @@ abstract class Task(taskContext : TaskContext, userConf : UserConfig) extends Ta
 
   implicit val self = taskContext.self
 
+  val taskConfig = userConf
```

Comment:
It is due to 
```
trait HBase {
  this: Task =>
...
  val pipelineConfig = taskConfig.getValue[PipelineConfig](PIPELINE)
```

An alternative would be

```
abstract class Task(taskContext : TaskContext, protected userConf : UserConfig) extends TaskInterface{
```

exposing the userConf to derived or self interfaces is valuable.